test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -248,123 +261,106 @@ private void CopyLatestCliToTestDirectory(string destinationDir)
         {
-            var packageFileExtractor = new PackageFileExtractor(packageFiles,
-                             PackageExtractionBehavior.XmlDocFileSaveMode);
-            var logger = new TestCommandOutputLogger();
-            var token = CancellationToken.None;
-            var filesCopied = new List<string>();
-
-            foreach (var packageFile in packageFiles)
+            var pathToPackSdk = Path.Combine(pathToSdkInCli, "Sdks", "NuGet.Build.Tasks.Pack");
+
+            const string packProjectName = "NuGet.Build.Tasks.Pack";
+            const string packTargetsName = "NuGet.Build.Tasks.Pack.targets";
+            // Copy the pack SDK.
+            var packProjectCoreArtifactsDirectory = new DirectoryInfo(Path.Combine(artifactsDirectory, packProjectName, toolsetVersion, "bin", configuration)).EnumerateDirectories("netstandard*").Single();
+            var packAssemblyDestinationDirectory = Path.Combine(pathToPackSdk, "CoreCLR");
+            // Be smart here so we don't have to call ILMerge in the VS build. It takes ~15s total.
+            // In VisualStudio, simply use the non il merged version.
+            var ilMergedPackDirectoryPath = Path.Combine(packProjectCoreArtifactsDirectory.FullName, "ilmerge");
+            if (Directory.Exists(ilMergedPackDirectoryPath))
             {
-                token.ThrowIfCancellationRequested();
-
-                var entry = nupkg.GetEntry(packageFile);
-
-                var packageFileName = entry.FullName;
-                // An entry in a ZipArchive could start with a '/' based on how it is zipped
-                // Remove it if present
-                if (packageFileName.StartsWith("/", StringComparison.Ordinal))
-                {
-                    packageFileName = packageFileName.Substring(1);
-                }
-                // Get only the name, without the path, since we are extracting to flat list
-                packageFileName = Path.GetFileName(packageFileName);
-
-                // ZipArchive always has forward slashes in them. By replacing them with DirectorySeparatorChar;
-                // in windows, we get the windows-style path
-                var normalizedPath = Uri.UnescapeDataString(packageFileName.Replace('/', Path.DirectorySeparatorChar));
-
-                var targetFilePath = Path.Combine(destination, normalizedPath);
-                if (!targetFilePath.StartsWith(destination, StringComparison.OrdinalIgnoreCase))
-                {
-                    continue;
-                }
-                try
+                var packFileName = packProjectName + ".dll";
+                // Only use the il merged assembly if it's newer than the build.
+                var packAssemblyCreationDate = new FileInfo(Path.Combine(packProjectCoreArtifactsDirectory.FullName, packFileName)).CreationTime;
+                var ilMergedPackAssemblyCreationDate = new FileInfo(Path.Combine(ilMergedPackDirectoryPath, packFileName)).CreationTime;
```

Comment:
[`File.GetCreationTimeUtc`](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.getcreationtimeutc?view=netframework-4.8) is more straightforward and doesn't create the throwaway `FileInfo` object.